sql/telemetry_derived/fission_monitoring_crash_v1/query.sql

```diff
@@ -0,0 +1,101 @@
+WITH crash_ping_data AS (
+  SELECT
+    submission_timestamp,
+    mozfun.map.get_key(environment.settings.user_prefs, 'fission.autostart') AS experiment_branch,
+    environment.build.build_id,
+    IF(payload.process_type = 'main' OR payload.process_type IS NULL, 1, 0) AS main_crash,
+    IF(
+      REGEXP_CONTAINS(payload.process_type, 'content')
+      AND NOT REGEXP_CONTAINS(COALESCE(payload.metadata.ipc_channel_error, ''), 'ShutDownKill'),
+      1,
+      0
+    ) AS content_crash,
+    IF(payload.metadata.startup_crash = '1', 1, 0) AS startup_crash,
+    IF(
+      REGEXP_CONTAINS(payload.process_type, 'content')
+      AND REGEXP_CONTAINS(payload.metadata.ipc_channel_error, 'ShutDownKill'),
+      1,
+      0
+    ) AS content_shutdown_crash,
+    -- 0 for values retrieved from main ping
+    0 AS usage_seconds,
+    0 AS gpu_crashes,
+    0 AS plugin_crashes,
+    0 AS gmplugin_crashes
+  FROM
+    `moz-fx-data-shared-prod.telemetry.crash`
+  WHERE
+    normalized_channel = 'nightly'
+    AND mozfun.map.get_key(environment.settings.user_prefs, 'fission.autostart') IS NOT NULL
+),
+main_ping_data AS (
+  SELECT
+    submission_timestamp,
+    mozfun.map.get_key(environment.settings.user_prefs, 'fission.autostart') AS experiment_branch,
+    environment.build.build_id,
+    0 AS main_crash,
+    0 AS content_crash,
```

Comment:
Please add `oom_crashes` and `shutdown_crashes`.

Reply:
@godelstheory how do I get them? i don't see them in the crash ping.

Reply:
For `oom_crashes` I have used:
```
 IF (payload.metadata.oom_allocation_size is not null,
      1,
      0) AS oom_crash**
```

For `shutdown_crashes`, maybe we should add three:
1. `shutdown_kill_crash`
```
payload.metadata.ipc_channel_error = 'ShutDownKill'
```
2. `shutdown_hang`
```
payload.metadata.moz_crash_reason LIKE 'MOZ_CRASH%'
```
3. `content_shutdown_crash`
```
IF (REGEXP_CONTAINS(payload.process_type, 'content')
   AND REGEXP_CONTAINS(payload.metadata.ipc_channel_error, 'ShutDownKill'),
       1,
       0) AS content_shutdown_crash,
```

Reply:
Adding oom_crashes and shutdown_hangs. As for the `shutdown_kill_crash` - defined this way (`payload.metadata.ipc_channel_error = 'ShutDownKill'` it will be a superset of `content_shutdown_crash` - is that expected?

Reply:
Yeah, I would include both.